packages/curve-ui-kit/src/shared/ui/Metric.tsx

```diff
@@ -0,0 +1,193 @@
+import Box from '@mui/material/Box'
+import Skeleton from '@mui/material/Skeleton'
+import Typography from '@mui/material/Typography'
+
+import { SizesAndSpaces } from 'curve-ui-kit/src/themes/design/1_sizes_spaces'
+import { TypographyVariantKey, TYPOGRAPHY_VARIANTS } from 'curve-ui-kit/src/themes/typography'
+import { abbr, suffix } from 'curve-ui-kit/src/utils'
+
+const { Spacing } = SizesAndSpaces
+
+// Correspond to flexbox align items values.
+export const ALIGNMENTS = ['start', 'center', 'end'] as const
+type Alignment = (typeof ALIGNMENTS)[number]
+
+const MetricSize = {
+  small: 'highlightM',
+  medium: 'highlightL',
+  large: 'highlightXl',
+  extraLarge: 'highlightXxl',
+} as const satisfies Record<string, TypographyVariantKey>
+
+export const SIZES = Object.keys(MetricSize) as (keyof typeof MetricSize)[]
+
+type UnitOptions = {
+  symbol: string
+  position: 'prefix' | 'suffix'
+  abbreviate: boolean
+}
+
+const dollar: UnitOptions = {
+  symbol: '$',
+  position: 'prefix',
+  abbreviate: true,
+}
+
+const percentage: UnitOptions = {
+  symbol: '%',
+  position: 'suffix',
+  abbreviate: false,
+}
+
+export const UNITS = ['dollar', 'percentage'] as const
+type Unit = (typeof UNITS)[number] | UnitOptions
+
+const UNIT_MAP: Record<(typeof UNITS)[number], UnitOptions> = {
+  dollar,
+  percentage,
+} as const
+
+// Default value formatter.
+const formatValue = (value: number, decimals?: number): string => {
+  return value.toLocaleString(undefined, {
+    minimumFractionDigits: decimals,
+    maximumFractionDigits: decimals,
+  })
+}
+
+// Default notional value formatter.
+const formatNotionalValue = (value: number, decimals?: number): string => {
+  return value.toLocaleString(undefined, {
+    minimumFractionDigits: decimals,
+    maximumFractionDigits: decimals,
+  })
+}
+
+const formatChange = (value: number) => {
+  // Looks aesthetically more pleasing without decimals.
+  if (value === 0) return '0'
+
+  return value.toLocaleString(undefined, {
+    minimumFractionDigits: 2,
+    maximumFractionDigits: 2,
+  })
+}
+
+type Props = {
+  /** Label that goes above the value */
+  label: string
+
+  /** The actual metric value to display */
+  value: number
+  /** Optional formatter for metric value */
+  formatter?: (value: number) => string
+  /** If the value should be abbreviated to 1.23k or 3.45m */
+  abbreviate?: boolean
+  /** The number of decimals the value should contain */
+  decimals?: number
```

Comment:
I think precision is favored for singular numbers like user deposits, but for a collection of numbers consistency is probably better? That's just my personal take on it though.

Reply:
yeah interesting take. I don't know how this will be used so we can leave it for now. It might still be worth to support precision in this component (when we need it?)